src/main/java/codeu/model/data/PrivateConversation.java

```diff
@@ -0,0 +1,99 @@
+/* PrivateConversation.java */
+/* Supports functionality to create private conversations, add users, and schedule 
+   surprises that correspond to adding another user in at a specified time */
+
+package codeu.model.data;
+
+import java.time.Instant; 
+import java.util.UUID; 
+import java.util.HashMap; 
+import java.util.ArrayList;
+import java.util.Timer; 
+import java.util.TimerTask;  
+import java.util.Date; 
+import java.util.Calendar; 
+
+/* Class representing a conversation, which can be thought of as a private chat room, created 
+   by a User, contains allowed Users, messages, and  future users to be added */
+public class PrivateConversation extends Conversation {
+
+    /* private class for tasks that send a notification and adds user to this conversation at scheduled time */
+    private class SurpriseTask extends TimerTask {
```

Comment:
So TimerTask is the class that we are using to figure out if it is time to put the user into the surprise?

Reply:
yup! Given a time, it will execute the "run" method at that time, so in this case, run() puts the user into the new gc.